src/Components/ResultLink.tsx

```diff
@@ -12,33 +16,25 @@ interface ILinkProps {
 }
 
 export default class ResultLink extends React.Component<ILinkProps, {}> {
+  private interactiveResult: InteractiveResult;
   private result: Result;
-  private wasOpened: Boolean;
 
   constructor(props: ILinkProps) {
     super(props);
     this.result = props.result;
-    this.wasOpened = false;
+    this.interactiveResult = buildInteractiveResult(headlessEngine, {
```

Comment:
I ended up changing this file quite a bit following you suggestion to use this controller @ThibodeauJF 
Could you please have a look through the file to make sure I didn't break something as a side effect?